### PR TITLE
Return only Store resources in Storefront API, Platform API and Storefront UI

### PR DESCRIPTION
**Pull request by damianlegawiec**
Storefront API, Platform API and Storefront will now return Store:
- [x] Products
- [x] Payment Methods
- [x] Orders

Resources from other stores will not be returned :)

TODO: 
- [x] add more tests for fetching store resources vs not accessing other stores resources
- [x] add more tests for `Product#for_store`
- [x] add more tests for `PaymentMethod#for_store`
- [x] add tests for  `Api::V2::Resourcecontroller#finder_params`
- [x] add more tests for Platform API as well
- [x] add tests for API v2 store switching

There will be required more work for Emails, but this will need to wait after https://github.com/spree/spree/pull/11110 is merged

Admin panel is handled in https://github.com/spree/spree/pull/11099

### CODE REVIEW

File: api/app/controllers/spree/api/v2/platform/resource_controller.rb [hunk 1]
```diff
@@ -61,7 +61,7 @@ def collection
           def scope
             return super if spree_current_user.present?
 
-            model_class.includes(scope_includes)
+            super(skip_cancancan: true)
```
Comment (mad-eel) [inline api/app/controllers/spree/api/v2/platform/resource_controller.rb:64]: Why are we skipping authorization when there is no current user? When does this case happen?

File: core/app/models/spree/store.rb [hunk 2]
```diff
@@ -47,9 +47,12 @@ class Store < Spree::Base
 
     alias_attribute :contact_email, :customer_support_email
 
-    def self.current(domain = nil)
-      current_store = domain ? Store.by_url(domain).first : nil
-      current_store || Store.default
+    def self.current(url = nil)
+      ActiveSupport::Deprecation.warn(<<-DEPRECATION, caller)
```
Comment (mad-eel) [inline core/app/models/spree/store.rb:51]: Shouldn't this warning be added to the `default` method?

Comment (damianlegawiec) [reply core/app/models/spree/store.rb:51]: @mad-eel nope, we only need to deprecate the `current` method as we've moved this logic to the finder class which can be easily swapped for a custom one by developers

File: api/app/controllers/spree/api/v2/platform/resource_controller.rb (hunk 1 above)
Comment (damianlegawiec) [reply api/app/controllers/spree/api/v2/platform/resource_controller.rb:64]: @mad-eel this is the platform API part - we're already authorizing app client/credentials here: https://github.com/spree/spree/blob/master/api/app/controllers/spree/api/v2/platform/resource_controller.rb#L10

Comment (damianlegawiec) [reply api/app/controllers/spree/api/v2/platform/resource_controller.rb:64]: we have 2 modes of using this API:

1. Apps - authorized by doorkeeper app token
2. Apps on behalf of users - authorized by doorkeeper app token + doorkeeper user token

**Review by mad-eel — APPROVED** [review]
Looks good overall 👍

**Review by MatthewKennedy — APPROVED** [review]
Looks good to me 👍🏻

**Pull request merged**